lib/rules/prefer-lookaround.ts

```diff
@@ -0,0 +1,633 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { CapturingGroup, Element } from "regexpp/ast"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+import { createTypeTracker } from "../utils/type-tracker"
+import type { KnownMethodCall, ReferenceElement } from "../utils/ast-utils"
+import {
+    getParent,
+    parseReplacements,
+    getStaticValue,
+    extractExpressionReferences,
+    isKnownMethodCall,
+} from "../utils/ast-utils"
+import type { PatternReplaceRange } from "../utils/ast-utils/pattern-source"
+import type { Expression, Literal } from "estree"
+import type { Rule } from "eslint"
+import { mention } from "../utils/mention"
+import { CharSetTreeRoot } from "../utils/char-set-tree"
+
+type ReplaceReference = { ref: string | number; range?: [number, number] }
```

Comment:
A range is given only if a string literal is given directly to the argument. If there is a range, it may be possible to autofix.

```js
'JavaScript.'.replace(/Java(Script)/g, 'Type$1')
```

However, if the argument string is given indirectly, no range is given. This will not be autofix.

```js
const replacement = 'Type'+'$1'
'JavaScript.'.replace(/Java(Script)/g, replacement)
```